modules/database/funcs.go

```diff
@@ -135,6 +134,9 @@ func (d *Database) CreateOrPatchDB(
 			return err
 		}
 
+		// If the service object doesn't have our finalizer, add it.
+		controllerutil.AddFinalizer(db, h.GetFinalizer())
```

Comment:
+1 I agree to hide the finalizer handling behind the Database abstraction in lib-common.